src/sagas/reown.js

```diff
@@ -0,0 +1,861 @@
+          });
+        };
+
+        dispatch({
+          type: types.SHOW_SIGN_MESSAGE_REQUEST_MODAL,
+          payload: {
+            accept: signMessageResponseTemplate(true),
+            deny: signMessageResponseTemplate(false),
+            data: request.data,
+            dapp: requestMetadata,
+          }
+        });
+      } break;
+
+      case TriggerTypes.SendNanoContractTxConfirmationPrompt: {
+        const sendNanoContractTxResponseTemplate = (accepted) => (data) => {
+          dispatch(hideGlobalModal());
+          resolve({
+            type: TriggerResponseTypes.SendNanoContractTxConfirmationResponse,
+            data: {
+              accepted,
+              nc: data
+            }
+          });
+        };
+
+        dispatch({
+          type: types.SHOW_NANO_CONTRACT_SEND_TX_MODAL,
+          payload: {
+            accept: sendNanoContractTxResponseTemplate(true),
+            deny: sendNanoContractTxResponseTemplate(false),
+            data: request.data,
+            dapp: requestMetadata,
+          }
+        });
+      } break;
+
+      case TriggerTypes.SendNanoContractTxLoadingTrigger:
+        dispatch(setNewNanoContractStatusLoading());
+        dispatch(showGlobalModal(MODAL_TYPES.NANO_CONTRACT_FEEDBACK, { isLoading: true }));
+        resolve();
+        break;
+
+      case TriggerTypes.CreateTokenLoadingTrigger:
+        dispatch(setCreateTokenStatusLoading());
+        resolve();
+        break;
+
+      case TriggerTypes.CreateTokenLoadingFinishedTrigger:
+        dispatch(setCreateTokenStatusReady());
+        resolve();
+        break;
+
+      case TriggerTypes.SendNanoContractTxLoadingFinishedTrigger:
+        dispatch(setNewNanoContractStatusReady());
+        resolve();
+        break;
+
+      case TriggerTypes.PinConfirmationPrompt: {
+        const pinPromise = new Promise((pinResolve, pinReject) => {
+          dispatch(showGlobalModal(MODAL_TYPES.PIN_PAD, {
+            onComplete: (pinCode) => {
+              dispatch(hideGlobalModal());
+              pinResolve(pinCode);
+            },
+            onCancel: () => {
+              dispatch(hideGlobalModal());
+              pinReject(new Error('PIN entry cancelled'));
+            }
+          }));
+        });
+
+        pinPromise
+          .then((pinCode) => {
+            resolve({
+              type: TriggerResponseTypes.PinRequestResponse,
+              data: {
+                accepted: true,
+                pinCode,
+              }
+            });
+          })
+          .catch(() => {
+            resolve({
+              type: TriggerResponseTypes.PinRequestResponse,
+              data: {
+                accepted: false,
+                pinCode: null,
+              }
+            });
+          });
+      } break;
+
+      default:
+        reject(new Error('Invalid request'));
+    }
+  });
+
+export function* onSignMessageRequest({ payload }) {
+  const { accept, deny: denyCb, data, dapp } = payload;
```

Comment:
question(non-blocking): Why not name the variable `acceptCb` to match `denyCb`?

Reply:
Done, thanks